lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -1872,6 +1872,12 @@ ThreadContext::PushEntryExitRecord(Js::ScriptEntryExitRecord * record)
         Assert(lastRecord->leaveForHost || lastRecord->leaveForAsyncHostOperation);
         lastRecord->hasReentered = true;
         record->next = lastRecord;
+
+        // these are on stack, which grows down. if this condition doesn't hold, then the list somehow got messed up
+        if ((uintptr_t)record > (uintptr_t)lastRecord)
+        {
+            RaiseFailFastException(NULL, NULL, NULL);
```

Comment:
There is no such convesion. You could see "RaiseFailFastException(nullptr, nullptr, 0);" in many similar projects.

---
In reply to: [62121674](https://github.com/Microsoft/ChakraCore/pull/923#discussion_r62121674) [](ancestors = 62121674)